### PR TITLE
Workshop notes

### PR DESCRIPTION
**Pull request by ahavrius**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
 
 ## Workshop description
 During the workshop you will learn what is Retrieval Augmented Generation, how it can increase trustability of LLM models and how to set up a RAG pipeline using Elastic.
 
 ## Requirements
-Do not forget to indicate Python version and any other tools
-+ add requirements.txt or conda.yml or docker image or Binder/Google Collab link
+
+Python version: >=3.8.5
+
```
Comment (nancyirisarri) [inline README.md:16]: Please add _requirements.txt_ or _conda.yaml_ or Binder/Google Collab link. Also, some users may not have Jupyter installed to start the notebook.

File: README.md [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
 
 ## Usage
 * Clone the repository
-* Start { TOOL } and navigate to the workshop folder
+* Start Visual Studio Code and navigate to the solutions folder
```
Comment (nancyirisarri) [inline README.md]: Please allow for other IDEs.

File: README.md (hunk 1 above)
Comment (nancyirisarri) [inline README.md:16]: Please add need to sign up at Elastic and the instructions (Settings) that should be used for the deployment.

Edit: I just saw you have this in chatbot.ipynb. Would it be okay to move it here and add the settings?

**Review by nancyirisarri — CHANGES_REQUESTED** [review]
Leaving remarks on notebooks separately, as not sure if can survive future commits.

For question-answering.ipynb:
- [x] Broken link for text _ElasticsearchStore_.
- [x] Broken link for text _OpenAIEmbeddings_.
- [x] Please add instructions for obtaining OpenAI keys.
- [x] Got _LangChainDeprecationWarning_ about _OpenAIEmbeddings_. I also got an error on the related cell, which seemed to be solved after fixing this. 
- [x] _LangChainDeprecationWarning_ about _ElasticsearchStore_.
- [x] Error 404 Not Found for https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/example-apps/workplace-search/data/data.json. Should use the local file instead perhaps?
- [x] After reading file, got message:
`Created a chunk of size 866, which is longer than the specified 800
Created a chunk of size 1120, which is longer than the specified 800`
- [x] _LangChainDeprecationWarning_ about _OpenAI_.

For chatbot.ipynb:
- [x] Got _LangChainDeprecationWarning_ about _OpenAIEmbeddings_ (same as in question-answering.ipynb).
- [x] _LangChainDeprecationWarning_ about _OpenAI_ (same as in question-answering.ipynb).
- [x] _LangChainPendingDeprecationWarning_ about _ElasticsearchChatMessageHistory_.

**Comment by gCaglia** [comment]
Agree with @nancyirisarri , but code-wise lgtm!
I will try and execute the full code including ElasticSearch Deployment shortly.

**Comment by gCaglia** [comment]
3 more points:
- If you are not planning it already, I would recommend starting with the ElasticSearch signup & deployment. That way we do not have to wait for the 5v minutes of deployment.
- Same goes for the OpenAI API Key; here, maybe we can have a breakout session before the actual talk, for everyone that does not have an API Key handy, but I will send you @nancyirisarri an E-Mail about that shortly.
- At the very end, at the end of the bottom of the chatbot.ipynb notebook, you @ahavrius suggest to experiment with more questions after clearing the workspace. Doing that breaks with a NotFoundError though. Is this expected behaviour? Either way I think it would be good to clarify here:
image

File: README.md (hunk 1 above)
Comment (ahavrius) [reply README.md:16]: done

File: README.md (hunk 2 above)
Comment (ahavrius) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (ahavrius) [reply README.md:16]: done

**Comment by ahavrius** [comment]
> 3 more points:
> 
> * If you are not planning it already, I would recommend starting with the ElasticSearch signup & deployment. That way we do not have to wait for the 5v minutes of deployment.
> * Same goes for the OpenAI API Key; here, maybe we can have a breakout session before the actual talk, for everyone that does not have an API Key handy, but I will send you @nancyirisarri an E-Mail about that shortly.
> * At the very end, at the end of the bottom of the chatbot.ipynb notebook, you @ahavrius suggest to experiment with more questions after clearing the workspace. Doing that breaks with a NotFoundError though. Is this expected behaviour? Either way I think it would be good to clarify here:
>   image

this is expected behaviour cause it doesn't find an index after running cells with cleaning index code. If you want to keep experimenting with a chat bot, please do it before cleaning up.

**Comment by ahavrius** [comment]
Thank you for your review  @nancyirisarri @GiuliaCaglia .  I resolved all the comments. Let me know if everything is ok

File: solutions/question-answering.ipynb [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        "llm = OpenAI(openai_api_key=OPENAI_API_KEY)\n",
+        "\n",
+        "qa = RetrievalQA.from_chain_type(\n",
+        "    llm=llm,\n",
+        "    chain_type=\"stuff\",\n",
+        "    retriever=retriever,\n",
+        "    return_source_documents=True\n",
+        ")\n",
+        "\n",
+        "ans = qa({\"query\": \"what is the nasa sales team?\"})\n",
```
Comment (gCaglia) [inline solutions/question-answering.ipynb]:
```suggestion
        "ans = qa.invoke({\"query\": \"what is the nasa sales team?\"})\n",
```

**Review by gCaglia — APPROVED** [review]
I have one additional suggestion to get rid of a deprecation error, but other than that, for me, it looks good!

File: solutions/question-answering.ipynb (hunk 3 above)
Comment (ahavrius) [reply solutions/question-answering.ipynb]: done

**Review by nancyirisarri — APPROVED** [review]
@ahavrius Just tried out installing the requirements and used the previous deployment and API key. Both notebooks run and look great. Thank you so much!

@GiuliaCaglia Let us see tomorrow how to tackle any OpenAI key issues. Valuable insight. Thank you as well!

**Pull request merged**